src/database/message-table.c

```diff
@@ -139,7 +160,7 @@ static bool add_message(enum message_type type,
 	}
 
 	// Ensure there are no duplicates when adding host name or rate-limiting messages
-	if(type == HOSTNAME_MESSAGE || type == RATE_LIMIT_MESSAGE)
+	if(unique)
```

Comment:
Yes, for instance one and the same regex error can happen for multiple regular expressions. Also `dnsmasq` warnings can pop up multiple times and it may be meaningful to keep them - like when reaching the maximum number of concurrently allowed upstream queries.